torchqdynamics/utils/tensor_types.py

```diff
@@ -67,6 +66,57 @@ def to_tensor(
         )
 
 
+def to_tdtensor(
+    x: TDOperatorLike,
+    dtype: torch.dtype | None = None,
+    device: torch.device | None = None,
+    is_complex: bool = False,
+):
+    """Convert a `TDOperatorLike` object to a `TDTensor` object."""
+    if is_complex:
+        dtype = cdtype(dtype)
+
+    if isinstance(x, get_args(OperatorLike)):
+        x = to_tensor(x, dtype=dtype, device=device, is_complex=True)
+    elif isinstance(x, callable):
+        # check number of arguments and compute at initial time
+        try:
+            x0 = x(0.0)
+        except TypeError:
+            raise TypeError(
+                'Time-dependent operators in the `callable` format should only accept a'
+                ' single argument for time `t`.'
+            )
```

Comment:
```suggestion
        except TypeError as e:
            raise TypeError(
                'Time-dependent operators in the `callable` format should only accept a'
                ' single argument for time `t`.'
            ) from e
```
Nitpicking, this is called exception chaining and is a best practice.

Reply:
Ok that's nice. Didn't know about this one.